### ISSUE TITLE
Level calculation is wrong

### ISSUE THREAD

**Issue by moepi2k** [suriyun-production/mmorpg-kit-docs#2616]
level formula is wrong. in game instance in default kit exp array is set 0-99 which means max level 100

but with it u can level to 101

https://gyazo.com/2388f396757a3299dfa6071886a89ee3

just the level from 100-101 dont visualy count the exp bar. it stays always at max. i just reduced exp table for the last 2 levels so u see that it still levels from 100 to 101

**Comment by insthync** [comment]
Nah, your understanding is wrong

Level start from 1 you can level up 99 times to reach level 100

The last array index is 99 so the array size is 100, so you can level up 100 times, so 1 + 100 = 101, that is it.

**Comment by moepi2k** [comment]
There is a bug believe. Check the video. And check the exp table calculator when u set max level to 100 the arrax limit is 98 which means u can level to 100. But there is a bug that on the last level it dont show when u get exo till u reach max exp for the very last level

**Comment by insthync** [comment]
from the image, your array size is 100, you can level up to 101.

**Comment by moepi2k** [comment]
Yes but for the last levem there is a visual bug on the exp bar. It not shows exp it seems.stuck till u reached last level

**Comment by insthync** [comment]
Yes, so all calculations are correct, right?

Only UI thing is wrong.

So what you've report is not correct.

**Comment by insthync** [comment]
From your image above, how did you count it to 98?

**Comment by moepi2k** [comment]
dont understand. i just made a new item table and set max level to 100 and click create. so it added the table with 0-98.

i just set the exp for the last to levels to low so that i can show you that there is an issue with the last level not showing exp on the exp bar

**Comment by insthync** [comment]
![Screenshot_251](https://github.com/user-attachments/assets/4a9e66a1-c5a9-4500-962c-719f2f761ce6)

From this image, the length is 100, so your character can level up 100 times so max level is 101
